### PR TITLE
CASSANDRA-19637(trunk): Makes conditions behavior on frozen and non-frozen columns consistent for null column values

### PR DESCRIPTION
**Pull request by blerer**
Thanks for sending a pull request! Here are some tips if you're new here:
 
 * Ensure you have added or run the [appropriate tests](https://cassandra.apache.org/_/development/testing.html) for your PR.
 * Be sure to keep the PR description updated to reflect all changes.
 * Write your PR title to summarize what this PR proposes.
 * If possible, provide a concise example to reproduce the issue for a faster review.
 * Read our [contributor guidelines](https://cassandra.apache.org/_/development/index.html)
 * If you're making a documentation change, see our [guide to documentation contribution](https://cassandra.apache.org/_/development/documentation.html)
 
Commit messages should follow the following format:

```
<One sentence description, usually Jira title or CHANGES.txt summary>

<Optional lengthier description (context on patch)>

patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####

Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>

```

The [Cassandra Jira](https://issues.apache.org/jira/projects/CASSANDRA/issues/)

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< HEAD
```
Comment (dcapwell) [inline CHANGES.txt:1]: merge conflict

File: src/java/org/apache/cassandra/db/Keyspace.java [hunk 2]
```diff
@@ -160,6 +160,14 @@ public static ColumnFamilyStore openAndGetStore(TableMetadata table)
         return open(table.keyspace).getColumnFamilyStore(table.id);
     }
 
+    public static ColumnFamilyStore openAndGetStoreIfExists(TableMetadata table)
```
Comment (dcapwell) [inline src/java/org/apache/cassandra/db/Keyspace.java:163]: Can we add `@Nullable`?

File: src/java/org/apache/cassandra/db/ColumnFamilyStore.java [hunk 3]
```diff
@@ -3315,13 +3315,16 @@ public static ColumnFamilyStore getIfExists(TableId id)
...
+    }
 
-        return keyspace.hasColumnFamilyStore(id)
-             ? keyspace.getColumnFamilyStore(id)
-             : null;
+    /**
+     * Returns a ColumnFamilyStore by metadata if it exists, null otherwise
+     * Differently from others, this method does not throw exception if the table does not exist.
+     */
+    public static ColumnFamilyStore getIfExists(TableMetadata table)
```
Comment (dcapwell) [inline src/java/org/apache/cassandra/db/ColumnFamilyStore.java:3325]: Can we add `@Nullable`?

File: src/java/org/apache/cassandra/db/Keyspace.java [hunk 4]
```diff
@@ -202,6 +210,11 @@ public ColumnFamilyStore getColumnFamilyStore(TableId id)
         return cfs;
     }
 
+    public ColumnFamilyStore getIfExists(TableId id)
```
Comment (dcapwell) [inline src/java/org/apache/cassandra/db/Keyspace.java:213]: Can we add `@Nullable`?

**Review by dcapwell — COMMENTED** [review]
don't know paxos well enough to comment that the changes are correct and safe, so would need someone like @belliottsmith to review that section.  The rest of the changes LGTM though

**Comment by blerer** [comment]
@dcapwell Looking at the PR I mixed up 2 patches here. Let me clean my mess.

**Pull request closed without merging**